tests/gcp/test_optimizers.py

```diff
@@ -100,3 +100,33 @@ def test_lbfgsb(generate_problem):
     assert isinstance(info, dict)
     assert not model.isequal(result)
     assert (model.full() - dense_data).norm() > (result.full() - dense_data).norm()
+
+
+def test_lbfgsb_callback(generate_problem):
+    dense_data, model, sampler = generate_problem
+
+    # Test default callback
+    maxiter = 2
+    solver = LBFGSB(maxiter=maxiter)
```

Comment:
The code has a comment about re-using the solver but we don't test for it. Can we add a test to make sure that is supported correctly? (My MR to fix the types also changes how we reset things when we are done so a test would confirm our intentions align)